src/Core/Utilities/AppHelpers.cs

```diff
@@ -437,8 +437,18 @@ public static bool SetAlternateMainPage(AppOptions appOptions)
                     App.MainPage = new NavigationPage(new CipherAddEditPage(appOptions: appOptions));
                     return true;
                 }
-                if (appOptions.FromFido2Framework && !string.IsNullOrWhiteSpace(appOptions.Fido2CredentialAction))
+
+                var userVerificationMediatorService = ServiceContainer.Resolve<IFido2MakeCredentialConfirmationUserInterface>();
+                if ((appOptions.FromFido2Framework && !string.IsNullOrWhiteSpace(appOptions.Fido2CredentialAction) || userVerificationMediatorService.IsWaitingUnlockVault))
                 {
+                    // If we are waiting for an unlock vault we don't want to trigger 'ExecuteFido2CredentialActionAsync' again,
+                    // as it's already running. We just need to 'ConfirmUnlockVault' on the 'userVerificationMediatorService'.
+                    if (userVerificationMediatorService.IsWaitingUnlockVault)
+                    {
+                       userVerificationMediatorService.ConfirmUnlockVault(true);
+                       return true;
+                    }
+
```

Comment:
🎨 To simplify this, couldn't we write it as:
```suggestion
                var userVerificationMediatorService = ServiceContainer.Resolve<IFido2MakeCredentialConfirmationUserInterface>();
                // If we are waiting for an unlock vault we don't want to trigger 'ExecuteFido2CredentialActionAsync' again,
                // as it's already running. We just need to 'ConfirmUnlockVault' on the 'userVerificationMediatorService'.
                if (userVerificationMediatorService.IsWaitingUnlockVault)
                {
                    userVerificationMediatorService.ConfirmUnlockVault();
                    return true;
                }
                
                if (appOptions.FromFido2Framework && !string.IsNullOrWhiteSpace(appOptions.Fido2CredentialAction))
                {
                    
```

Reply:
Yes, looks like it. Just like the one of the other scenarios above it's remnants of older logic that I had.